eland/ml/exporters/_sklearn_deserializers.py

```diff
@@ -0,0 +1,210 @@
+#  the Apache License, Version 2.0 (the "License"); you may
+#  not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+import numpy as np
+import sklearn
+from sklearn.preprocessing import FunctionTransformer
+
+
+class Tree:
+    """Wrapper to create sklearn Tree objects from Elastic ML tree
+    description in JSON format.
+    """
+
+    def __init__(self, json_tree, feature_names_map):
+        TREE_LEAF = -1
+
+        node_count = len(json_tree["tree_structure"])
+        children_left = np.ones((node_count,), dtype=int) * TREE_LEAF
+        children_right = np.ones((node_count,), dtype=int) * TREE_LEAF
+        feature = np.ones((node_count,), dtype=int) * -2
+        threshold = np.ones((node_count,), dtype=float) * -2
+        impurity = np.zeros((node_count,), dtype=float)
+        # value works only for regression and binary classification
+        value = np.zeros((node_count, 1, 1), dtype="<f8")
+        n_node_samples = np.zeros((node_count,), dtype=int)
+
+        # parse values from the JSON tree
+        feature_names = json_tree["feature_names"]
+        for json_node in json_tree["tree_structure"]:
+            node_id = json_node["node_index"]
+            if "number_samples" in json_node:
+                n_node_samples[node_id] = json_node["number_samples"]
+            else:
+                n_node_samples[node_id] = 0
+
+            if "leaf_value" not in json_node:
+                children_left[node_id] = json_node["left_child"]
+                children_right[node_id] = json_node["right_child"]
+                feature[node_id] = feature_names_map[
+                    feature_names[json_node["split_feature"]]
+                ]
+                threshold[node_id] = json_node["threshold"]
+                if "split_gain" in json_node:
+                    impurity[node_id] = json_node["split_gain"]
+                else:
+                    impurity[node_id] = -1
+            else:
+                value[node_id, 0, 0] = json_node["leaf_value"]
+
+        # iterate through tree to get max depth and expected values
+        weighted_n_node_samples = n_node_samples.copy()
+        self.max_depth = Tree._compute_expectations(
+            children_left=children_left,
+            children_right=children_right,
+            node_sample_weight=weighted_n_node_samples,
+            values=value,
+            node_index=0,
+        )
+        self.n_outputs = value.shape[-1]
+
+        # initialize the sklearn tree
+        self.tree_ = sklearn.tree._tree.Tree(
```

Comment:
I guess if other packages rely on them then they've kind of leaked. My main concern is we'd might hit problems with sklearn versions. It might be worth adding your comment here as a code comment. Long term I guess we may have version dependent code here as well :(.